### PR TITLE
Add unit tests for cilium load balancer

### PR DESCRIPTION
**Pull request by HomayoonAlimohammadi**
### Summary
- Add unit tests for `Cilium load-balancer` k8sd feature
- Rename some `Cilium network` tests and functions to prevent conflict

### CODE REVIEW

**Comment by HomayoonAlimohammadi** [comment]
the reason I implemented new k8s mock was the discovery, I noticed that I can use the fake discovery. closing this to re-open with new changes.

File: src/k8s/pkg/k8sd/features/cilium/loadbalancer_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+		deployment, err := clientset.AppsV1().Deployments("kube-system").Get(context.Background(), "cilium-operator", metav1.GetOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(deployment.Spec.Template.Annotations).To(HaveKey("kubectl.kubernetes.io/restartedAt"))
+		daemonSet, err := clientset.AppsV1().DaemonSets("kube-system").Get(context.Background(), "cilium", metav1.GetOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(daemonSet.Spec.Template.Annotations).To(HaveKey("kubectl.kubernetes.io/restartedAt"))
+	})
+}
+
+func validateLBValues(t *testing.T, values map[string]interface{}, lbCfg types.LoadBalancer) {
```
Comment (bschimke95) [inline src/k8s/pkg/k8sd/features/cilium/loadbalancer_test.go:215]: nit: in the metallb test you call this `validateLoadBalancerValues`. Let's be consistent on either option

**Review by bschimke95 — APPROVED** [review]
LGTM, only a super small nit.

**Review by eaudetcobello — APPROVED** [review]
Nice work !

**Pull request merged**